scripts/vendor-boringssl.sh

```diff
@@ -132,6 +132,41 @@ function mangle_symbols {
     namespace_inlines "$DSTROOT"
 }
 
+
+# BoringSSL includes a few non-namespaced C++ structures. These aren't namespaced because they're exposed
+# in C-land, which doesn't know about the namespacing. Sadly, these structures include constructors and destructors,
+# and if those aren't namespaced we're still able to conflict.
+#
+# This function is responsible for identifying them and manually cleaning them up. We run this only on
+# macOS because we don't believe that the cross-platform architectures will hit any other structures.
```

Comment:
probably silly question, but why?

Reply:
The only way this would happen is if there were structures present that met the following criteria:

1. Names exported into C interface files (i.e. public headers), so they can't be namespaced
2. But only on some platforms

This is fairly unlikely: BoringSSL doesn't have that much platform specific code, and even less platform-specific API.